testing/scenario_app/android/README.md

```diff
@@ -1,102 +1,364 @@
+>   `ExternalTextureFlutterActivity` sometimes do not render.
+> - [#144232](https://github.com/flutter/flutter/issues/144232): Impeller Vulkan
+>   sometimes hangs on emulators.
+> - [#144352](https://github.com/flutter/flutter/issues/144352): Skia Gold
+>   sometimes is missing expected diffs.
+
+---
+
+This test suite was [originally written in 2019](https://github.com/flutter/engine/pull/10007)
+with a goal of:
+
+> \[being\] suitable for embedders to do integration testing with - it has no
+> dependencies on the flutter_tools or framework, and so will not fail/flake
+> based on variances in those downstream.
+
+Unfortunately, the Android side of the test suite was never fully operational,
+and the tests, even if failing, were accidentally be reported as passing on CI.
+In 2024, as the team got closer to shipping our new graphics backend,
+[Impeller](https://docs.flutter.dev/perf/impeller) on Android, it was clear that
+we needed a reliable test suite for the engine on Android, particularly for
+visual tests around external textures and platform views.
+
+So, this package was revived and updated to be a (more) reliable test suite for
+the engine on Android. It's by no means complete
+([contributions welcome](#contributing)), but it did successfully catch at least
+one bug that would not have been detected automatically otherwise.
+
+_Go forth and test the engine on Android!_
+
+---
+
+Top topics covered in this document include (but are not limited to):
+
+- [Running the Tests](#running-the-tests)
+- [Contributing](#contributing)
+- [Troubleshooting](#troubleshooting)
+- [Getting Help](#getting-help)
+
+## Introduction
+
+This package simulates a Flutter app that uses the engine (`dart:ui`) only, in
+conjunction with Android-specific embedding code that simulates the use of the
+engine in a real app (such as plugins and platform views).
+
+A custom test runner, [`run_android_tests.dart`](bin/run_android_tests.dart), is
+used to run the tests on a connected device or emulator, and report the results,
+including screenshots (golden files) and logs from `adb logcat`.
+
+In the following architecture diagram:
+
+> <a
+>   href="https://github.com/flutter/flutter/wiki/The-Engine-architecture"
+>   title="The Engine architecture"> <img
+>   width="300"
+>   alt="Anatomy of a Flutter app"
+>   src="https://raw.githubusercontent.com/flutter/engine/main/docs/app_anatomy.svg"
+> /> </a>
+
+- The Dart app is represented by [`lib/main.dart`](../lib/main.dart).
+- There is no framework code.
+- `dart:ui` and the engine are the same as in a real app.
+- Android-specific application code is in this directory
+  ([`android/`](./)).
+- The runner is a _custom_ test runner,
+  [`run_android_tests.dart`](bin/run_android_tests.dart).
+
+### Scope of Testing
+
+The tests in this package are end-to-end tests that _specifically_ exercise the
+engine and Android-specific embedding code. They are not unit tests for the
```

Comment:
It would be nice to add a link after "unit tests" to the link provided below.  Maybe using something like `[1]`

Reply:
Done!